### PR TITLE
Operationalize Vagrant

### PR DESCRIPTION
**Pull request by qwwqwwq**
- Fixes the packer config to install the correct version of node.
- Adds script for running all tests.
- Updates readme with instruction on using vagrant.
- Remove unneeded mognodb/redis startup scripts.
- Make app.js executable
- Setup maven wrapper for microservice

### CODE REVIEW

File: winterfell/SETUP_WINTERFELL.md [hunk 1]
```diff
@@ -1,58 +1,48 @@
...
 $ npm install
 ```
 
-### Install MongoDb No-SQL Database
-Refer to the [MongoDb installation page](https://docs.mongodb.com/master/installation/) to install.
-We used Homebrew to install Mongo.
-```
-$ brew install mongodb
-```
+### Startup Vagrant
```
Comment (ahsu1230) [inline winterfell/SETUP_WINTERFELL.md]: Please include a short blurb with how to Download Vagrant and virtualbox + any other things to set it up correctly before moving on to "vagrant up"

File: winterfell/SETUP_WINTERFELL.md [hunk 2]
```diff
@@ -1,58 +1,48 @@
...
+### Run Tests
 
-Make sure a _Redis_ instance is started in a separate terminal, we assume the database is on port 6379.
-```
-$ redis-server
-```
-or
 ```
-$ ./start-redis.sh
+$ vagrant ssh
```
Comment (ahsu1230) [inline winterfell/SETUP_WINTERFELL.md]: can we have a script in the winterfell directory ./run-test.sh that 
calls vagrant ssh, and then runs the tests without having to change the current directory? so like ./vagrant/winterfell/run-tests.sh
that way we can stay on the winterfell directory without having to go back and forth

File: winterfell/SETUP_WINTERFELL.md [hunk 3]
```diff
@@ -1,58 +1,48 @@
...
+$ cd /vagrant/winterfell
+$ ./run-tests.sh
 ```
 
-Simply run the _start.sh_ script from the command line. The server should start up at port 3999.
+### Run Server
+
 ```
+$ vagrant ssh
```
Comment (ahsu1230) [inline winterfell/SETUP_WINTERFELL.md]: same here.
Let's have a script run-server.sh in the winterfell directory that calls
vagrant ssh
./vagrant/winterfell/start-server.sh

This keeps us at our current working directory.

Comment (ahsu1230) [inline winterfell/SETUP_WINTERFELL.md]: oh wait nvm I see how it works, hmm

File: winterfell/SETUP_WINTERFELL.md (hunk 2 above)
Comment (ahsu1230) [inline winterfell/SETUP_WINTERFELL.md]: you can ignore this comment as well

File: config/packer/scripts/dep.sh [hunk 4]
```diff
@@ -14,8 +14,8 @@ export DEBIAN_FRONTEND=noninteractive
 # To install the add-apt-repository utility
 apt-get -y install software-properties-common python-software-properties
 
-# Node
-curl -sL https://deb.nodesource.com/setup | bash -
+# Node 4.x
```
Comment (ahsu1230) [inline config/packer/scripts/dep.sh]: 5.x

File: winterfell/SETUP_WINTERFELL.md [hunk 5]
```diff
@@ -1,58 +1,48 @@
 # Setting up Winterfell
 Winterfell is the server that handles all control logic in the application. This includes authentication, handling user information, it is basically the bulk of the main logistics.
 
-### Install Node
-This project is designed to work with node.js v5. 
+### Install Node Modules
+This project is designed to work with node.js v4.x.
```
Comment (ahsu1230) [inline winterfell/SETUP_WINTERFELL.md]: v5.x

File: winterfell/pdf-filler/.mvn/wrapper/maven-wrapper.properties [hunk 6]
```diff
@@ -0,0 +1 @@
+distributionUrl=https://repo1.maven.org/maven2/org/apache/maven/apache-maven/3.3.3/apache-maven-3.3.3-bin.zip
```
Comment (ahsu1230) [inline winterfell/pdf-filler/.mvn/wrapper/maven-wrapper.properties]: I don't think we want .mvn contents in our repo.
We should add it to our git ignore

Comment (qwwqwwq) [inline winterfell/pdf-filler/.mvn/wrapper/maven-wrapper.properties]: These files are required for the functionality of the maven wrapper, they are designed to be checked into source control. Mvn wrapper allows you to run a consistent version of maven via mvnw, its more reliable then assuming mvn in installed on the local system

**Pull request merged**